core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectOrchestrator.scala

```diff
@@ -1,33 +1,48 @@
 import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
 
-import java.net.URLEncoder
-import java.nio.charset.StandardCharsets.UTF_8
 import scala.concurrent.{ExecutionContext, Future}
 
 
 /**
- * Knows how to provision billing projects with external cloud providers
+ * Knows how to provision billing projects with external cloud providers (that is, implementors of the
+ * BillingProjectCreator trait)
+ *
+ * All billing projects are created following this algorithm:
+ * 1. Pre-flight validation with a billing project creator. Right now, the creator is determined according
+ * to the nature of the billing project creation request and is not client-configurable.
+ * 2. Create the rawls internal billing project record
+ * 3. Post-flight steps; this may include syncing of groups, reaching out to external services to sync state, etc.
+ * This step is delegated to the billing project creator as well.
  */
-class BillingProjectOrchestrator(userInfo: UserInfo, samDAO: SamDAO, gcsDAO: GoogleServicesDAO, billingRepository: BillingRepository)(implicit val executionContext: ExecutionContext) extends StringValidationUtils {
+class BillingProjectOrchestrator(userInfo: UserInfo,
+                                 samDAO: SamDAO,
+                                 billingRepository: BillingRepository,
+                                 googleBillingProjectCreator: BillingProjectCreator,
+                                 bpmBillingProjectCreator: BillingProjectCreator)
+                                (implicit val executionContext: ExecutionContext) extends StringValidationUtils {
   implicit val errorReportSource = ErrorReportSource("rawls")
 
+  /**
+   * Creates a "v2" billing project, using either Azure managed app coordinates or a Google Billing Account
+   */
   def createBillingProjectV2(createProjectRequest: CreateRawlsV2BillingProjectFullRequest): Future[Unit] = {
+    val billingProjectCreator = createProjectRequest.billingInfo match {
+      case Left(_) => googleBillingProjectCreator
+      case Right(_) => bpmBillingProjectCreator
+    }
+
     for {
       _ <- validateBillingProjectName(createProjectRequest.projectName.value)
-      _ <- ServicePerimeterService.checkServicePerimeterAccess(samDAO, createProjectRequest.servicePerimeter, userInfo)
-      hasAccess <- gcsDAO.testBillingAccountAccess(createProjectRequest.billingAccount, userInfo)
-      _ = if (!hasAccess) {
-        throw new GoogleBillingAccountAccessException(ErrorReport(StatusCodes.BadRequest, "Billing account does not exist, user does not have access, or Terra does not have access"))
-      }
+
+      _ <- billingProjectCreator.validateBillingProjectCreationRequest(createProjectRequest, userInfo)
       result <- createV2BillingProjectInternal(createProjectRequest, userInfo)
+      _ <- billingProjectCreator.postCreationSteps(createProjectRequest, userInfo)
```

Comment:
Add a rollback step if this fails

Reply:
Implemented in [this commit](https://github.com/broadinstitute/rawls/pull/1961/commits/ba8df16cab10823c9f6f1e92a36ac02416aa8d23)